### PR TITLE
Restructure and Refactor

### PR DESCRIPTION
**Pull request by HellenNamulinda**
This PR is for the CLI implementation.

The Training Workflow, 
the train method
- Loads input data and splits it into training and validation sets.
- Determines if the task is binary classification or regression based on the target variable.
- Selects the appropriate molecular descriptor based on user input.
- Fits the descriptor to the training data and transforms the data into feature representations.
- Initializes the appropriate model (classifier or regressor) and trains it on the training data.
- Evaluates the model on the validation set and generates explanations.
- Retrains the final model on the entire dataset and saves it.
- Save the descriptor in the same output directory.

Inference Workflow:
- Loads the trained model, descriptor and input data.
- Transforms the input data using the same descriptor used during training.
- Save prediction results as a csv file.
- Generate and reports(distribution of predictions and interpretability plots).

### CODE REVIEW

**Comment by miquelduranfrigola** [comment]
This looks good so far. Thanks @HellenNamulinda

**Comment by HellenNamulinda** [comment]
Hello @miquelduranfrigola,
This is ready for merging.
